pisound-btn/pisound-btn.c

```diff
@@ -300,37 +301,37 @@ static void get_action_script(enum action_e action, char* action_name, char *buf
 	read_config_value(g_config_path, action_name, buffer, args, buff_length, "#");
 	debug( 2, "read_config_value returned %s\n", buffer );
 	// if buffer[0] == '#' then the entry did not exist in the config file.
-	if (buffer[0] == '#')// entry was not specified -- use defaults.
+	if (buffer[0] == '#' && g_use_default)// entry was not specified -- use defaults.
 	{
 		bool found = false;
 		switch (action)
 		{
 		case A_UP:
-			strcpy( buffer, DEFAULT_UP );
+			strncpy( buffer, DEFAULT_UP, MAX_PATH_LENGTH );
```

Comment:
MAX_PATH_LENGTH should instead be buff_length-1, and just in case `buffer[buff_length-1] = '\0';` should be added to ensure the contents in `buffer` are null terminated. (The null character does not get added in case the string length exceeds length argument)

However, when copying predefined built in constant strings, it's safe to assume the copied values are going to be of reasonable length and `strcpy` is fine in all these places. A sufficient addition to this function would be to check at the beginning whether `buff_length` provided is of big enough value to at least store the longest default value, or even longer values.

Reply:
Changes now look ok, but one `strncpy` unrelated to previous changes got modified. I would fix it myself and merge it, but I'm not sure there's a straightforward way for me to put in code changes in the pull request... Marked it in another 'review' comment.